flake.nix

```
@@ -1,53 +1,69 @@
 {
   description = "Nix package for centrifuge-chain";
 
-  inputs.nixpkgs.url = github:NixOS/nixpkgs/nixos-20.09;
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-20.09;
+    gitignore-nix = {
+      url = github:hercules-ci/gitignore.nix;
+      flake = false;
+    };
+  };
 
-  outputs = { self, nixpkgs, ... }: let
-    name = "centrifuge-chain";
-    version = "2.0.0";
-  in
+  outputs = inputs:
+    let
+      name = "centrifuge-chain";
+      version = "2.0.0";
+      pkgs = inputs.nixpkgs.legacyPackages.x86_64-linux;
+      gitignore = (
+        import inputs.gitignore-nix {
+          inherit (inputs.nixpkgs.legacyPackages.x86_64-linux) lib;
+        }
+      ).gitignoreSource;
+    in
     {
-      defaultPackage.x86_64-linux =
-        with import nixpkgs { system = "x86_64-linux"; };
-
-        rustPlatform.buildRustPackage {
+      packages.x86_64-linux.centrifuge-chain =
+        pkgs.rustPlatform.buildRustPackage {
           pname = name;
           version = version;
 
-          src = self;
+          src = gitignore ./.;
 
           cargoSha256 = "sha256-52CN7N9FQiJSODloo0VZGPNw4P5XsaWfaQxEf6Nm2gI=";
 
-          nativeBuildInputs = [ clang pkg-config ];
-          buildInputs = [ openssl ];
+          nativeBuildInputs = with pkgs; [ clang pkg-config ];
+          buildInputs = with pkgs; [ openssl ];
 
-          LIBCLANG_PATH = "${llvmPackages.libclang}/lib";
-          PROTOC = "${protobuf}/bin/protoc";
+          LIBCLANG_PATH = "${pkgs.llvmPackages.libclang}/lib";
+          PROTOC = "${pkgs.protobuf}/bin/protoc";
           BUILD_DUMMY_WASM_BINARY = 1;
 
           doCheck = false;
         };
 
-      packages.x86_64-linux.dockerContainer = let
-        pkgs = import nixpkgs { system = "x86_64-linux"; };
-      in
+      defaultPackage.x86_64-linux = inputs.self.packages.x86_64-linux.centrifuge-chain;
+
+      packages.x86_64-linux.dockerContainer =
         pkgs.dockerTools.buildImage {
           name = "centrifugeio/${name}";
           tag = "latest";
 
+          contents = inputs.self.defaultPackage.x86_64-linux;
+
           config = {
+            Env = [
+              "PATH=/bin/centrifuge-chain"
+            ];
             ExposedPorts = {
-              "30333/tcp" = {};
-              "9933/tcp" = {};
-              "9944/tcp" = {};
+              "30333/tcp" = { };
+              "9933/tcp" = { };
+              "9944/tcp" = { };
             };
             Volumes = {
-                "/data" = {};
+              "/data" = { };
             };
-            Cmd = [ "${self.defaultPackage.x86_64-linux}/bin/centrifuge-chain" ];
+            Entrypoint = [ "/bin/centrifuge-chain" ];
```

Can we use here the binary name only without the `/bin`, since it would be in the path after the above is adjusted.
```suggestion
            Entrypoint = [ "centrifuge-chain" ];
```

---

Done.